src/analysis/processing/qgsalgorithmrastersampling.cpp

```diff
@@ -0,0 +1,166 @@
+{
+  return QObject::tr( "Sample raster values" );
+}
+
+QStringList QgsRasterSamplingAlgorithm::tags() const
+{
+  return QObject::tr( "extract,point,pixel,value" ).split( ',' );
+}
+
+QString QgsRasterSamplingAlgorithm::group() const
+{
+  return QObject::tr( "Raster analysis" );
+}
+
+QString QgsRasterSamplingAlgorithm::groupId() const
+{
+  return QStringLiteral( "rasteranalysis" );
+}
+
+QString QgsRasterSamplingAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Samples raster values under a set of points." );
+}
+
+QString QgsRasterSamplingAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm creates a new vector layer with the same attributes of the input layer and the raster values corresponding on the point location.\n\n"
+                      "If the raster layer has more than one band, all the band values are sampled." );
+}
+
+QgsRasterSamplingAlgorithm *QgsRasterSamplingAlgorithm::createInstance() const
+{
+  return new QgsRasterSamplingAlgorithm();
+}
+
+void QgsRasterSamplingAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  addParameter( new QgsProcessingParameterRasterLayer( QStringLiteral( "RASTERCOPY" ), QObject::tr( "Raster layer" ) ) );
+
+  addParameter( new QgsProcessingParameterString( QStringLiteral( "COLUMN_PREFIX" ), QObject::tr( "Output column prefix" ), QStringLiteral( "SAMPLE_" ), false, true ) );
+  addParameter( new QgsProcessingParameterFeatureSink( QStringLiteral( "OUTPUT" ), QObject::tr( "Sampled" ), QgsProcessing::TypeVectorPoint ) );
+}
+
+bool QgsRasterSamplingAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  QgsRasterLayer *layer = parameterAsRasterLayer( parameters, QStringLiteral( "INPUT_RASTER" ), context );
+  if ( !layer )
+    throw QgsProcessingException( invalidRasterError( parameters, QStringLiteral( "INPUT_RASTER" ) ) );
+
+  mBandCount = layer->bandCount();
+  mCrs = layer->crs();
+  mDataProvider.reset( static_cast< QgsRasterDataProvider *>( layer->dataProvider()->clone() ) );
+
+  return true;
+}
+
+QVariantMap QgsRasterSamplingAlgorithm::processAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  std::unique_ptr< QgsFeatureSource > source( parameterAsSource( parameters, QStringLiteral( "INPUT" ), context ) );
+  if ( !source )
+    throw QgsProcessingException( invalidSourceError( parameters, QStringLiteral( "INPUT" ) ) );
+
+  QString fieldPrefix = parameterAsString( parameters, QStringLiteral( "COLUMN_PREFIX" ), context );
+  QgsFields newFields;
+  for ( int band = 1; band <= mBandCount; band++ )
+  {
+    newFields.append( QgsField( QStringLiteral( "%1%2" ).arg( fieldPrefix, QString::number( band ) ), QVariant::Double ) );
```

Comment:
If we wanted to be cleverer, we could check the raster data type and use an int field here when appropriate.